lib/screens/v2/widget_instance/reconstructed_alert.ex

```diff
@@ -90,14 +94,32 @@ defmodule Screens.V2.WidgetInstance.ReconstructedAlert do
       |> hd()
 
     if is_nil(direction_id) do
-      nil
+      informed_stop_ids = Enum.into(informed_entities, MapSet.new(), & &1.stop)
+
+      :screens
+      |> Application.get_env(:reconstructed_alert_headsign_matchers)
+      |> Map.get(stop_id)
+      |> Enum.find_value(nil, fn {informed, not_informed, headsign} ->
+        if alert_region_match?(to_set(informed), to_set(not_informed), informed_stop_ids),
+          do: headsign,
+          else: false
+      end)
```

Comment:
Something to consider is pulling this out into some common file for this widget and DUPs. The code looks the same so would be less overhead to have it in one place. Up to you though.